src/reliability_utils.nim

```diff
@@ -0,0 +1,95 @@
+import std/[times, locks]
+import ./[rolling_bloom_filter, message]
+
+type
+  PeriodicSyncCallback* = proc() {.gcsafe, raises: [].}
+
+  ReliabilityConfig* = object
+    bloomFilterCapacity*: int
+    bloomFilterErrorRate*: float
+    bloomFilterWindow*: times.Duration
```

Comment:
I argue elsewhere that we shouldn't bring `Duration` into the bloom filter design (as only _number_ of inserted items affect the false positive design and message rates over specific time periods are extremely variable).